e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+
+    def expect_button_label(
+        self,
+        button_label: TextValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if button_label is None:
+            button_label = ""
+        playwright_expect(self.loc_button).to_have_text(button_label, timeout=timeout)
+
+    def expect_capture(
+        self,
+        capture: typing.Union[Literal["environment", "user"], None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc, "capture", capture, timeout=timeout)
+
+    def expect_placeholder_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc_file_display, "placeholder", value=value, timeout=timeout)
+
+
+class InputSlider(_WidthLoc, _InputWithLabel):
+    # id: str,
+    # label: TagChildArg,
+    # min: SliderValueArg,
+    # max: SliderValueArg,
+    # value: Union[SliderValueArg, Iterable[SliderValueArg]],
+    # step: Optional[SliderStepArg] = None,
+    # ticks: bool = True,
+    # animate: Union[bool, AnimationOptions] = False,
+    # width: Optional[str] = None,
+    # sep: str = ",",
+    # pre: Optional[str] = None,
+    # post: Optional[str] = None,
+    # time_format: Optional[str] = None,
+    # timezone: Optional[str] = None,
+    # drag_range: bool = True,
+
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"input#{id}",
+            loc_label=f"label#{id}-label",
+        )
+
+    def expect_value(self, value: AttrValue, *, timeout: Timeout = None) -> None:
+        # TODO-barret; implement
+        NotImplementedError("Need to get the value somehow")
+
+    def set(self, fraction: float, *, timeout: Timeout = None) -> None:
+        if fraction > 1 or fraction < 0:
+            raise ValueError("`fraction` must be between 0 and 1")
+
+        self.loc_container.wait_for(state="visible", timeout=timeout)
+        self.loc_container.scroll_into_view_if_needed(timeout=timeout)
+
+        handle = self.loc_container.locator(".irs-handle")
+        handle_bb = handle.bounding_box(timeout=timeout)
+        if handle_bb is None:
+            raise RuntimeError("Couldn't find bounding box for .irs-handle")
+
+        handle_center = (
+            handle_bb.get("x") + (handle_bb.get("width") / 2),
+            handle_bb.get("y") + (handle_bb.get("height") / 2),
+        )
+
+        grid = self.loc_container.locator(".irs-grid")
+        grid_bb = grid.bounding_box(timeout=timeout)
+        if grid_bb is None:
+            raise RuntimeError("Couldn't find bounding box for .irs-grid")
+
+        mouse = self.loc_container.page.mouse
+        mouse.move(handle_center[0], handle_center[1])
+        mouse.down()
+        mouse.move(
+            grid_bb.get("x") + (fraction * grid_bb.get("width")), handle_center[1]
+        )
+        mouse.up()
```

Comment:
To implement "set value", simulate what the user would do: move the mouse and look at the handle value to see if it matches the desired value. (Ideally we'd do this with a single mouse.down()/mouse.up(), just lots of mouse.move())

Reply:
a739fe3

Reply:
This is a satisfying solution. Much more intuitive